app/assets/templates/header_tpl.jst.hbs

```diff
@@ -1,109 +1,99 @@
+            </li>
+            <li id="view-all" role="presentation">
+              <a href="/notifications" id="view_all_notifications">{{t "header.view_all"}}</a>
+            </li>
+          </ul>
+        </div>
+      </li>
 
-    <div class="badge badge-inverse" id="conversations_badge">
-     <div class="icons-mail_grey" >
-        <a href="/conversations" title="{{t "header.conversations"}}" class="badge_link" >
+      <li>
+        <a href="/conversations" title="{{t "header.conversations"}}">
+          <i class="entypo mail"></i>
           <div class="badge_count {{#unless current_user.unread_messages_count}} hidden {{/unless}}">
             {{current_user.unread_messages_count}}
           </div>
         </a>
-      </div>
-    </div>
+      </li>
+    </ul>
 
+    <div id="user-menu" class="dropdown header-dropdown">
+      <button class="dropdown-toggle" type="button" data-toggle="dropdown">
+        <img class="user-menu-avatar" alt="{{current_user.name}}"
+          src="{{current_user.avatar.small}}" title="{{current_user.name}}"/>
+        <span id="user-id">{{current_user.name}}</span>
+        <span id="caret">▼</span>
+      </button>
+      <ul class="dropdown-menu pull-right" role="menu">
+        <!--
+          !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+          !  Class class="dropdown-menu pull-right" will have to be !
+          !  replaced by "dropdown-menu-right with bootstrap 3 !    !
+          !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        -->
+        <li role="presentation"><a href="/people/{{current_user.guid}}">{{t "header.profile"}}</a></li>
+        <li role="presentation"><a href="/contacts">{{t "header.contacts"}}</a></li>
+        <li role="presentation"><a href="/user/edit">{{t "header.settings"}}</a></li>
+        <li role="presentation"><a href="/help">{{t "header.help"}}</a></li>
+        {{#if current_user.admin}}
+        <li role="presentation"><a href="/admins/user_search">{{t "header.admin"}}</a></li>
+        {{/if}}
+        <li role="presentation"><a href="/users/sign_out" data-method="delete">{{t "header.log_out"}}</a></li>
+      </ul>
+    </div>
 
-      <div id="notification_dropdown">
-        <div class="header">
-          <div class="pull-right">
-            <a href="#" id="mark_all_read_link" class="{{#unless current_user.notifications_count}}disabled{{/unless}}">
-              {{t "header.mark_all_as_read"}}
-            </a>
-          </div>
-
-          <h4>
-            {{t "header.recent_notifications"}}
-          </h4>
-        </div>
+    <div id="global_search">
+      <form accept-charset="UTF-8" action="/search" class="search_form" method="get">
+        <input name="utf8" type="hidden" value="✓">
+        <input id="q" name="q" placeholder="{{t "header.search"}}" results="5" type="search" autocomplete="off" class="ac_input">
+      </form>
+    </div>
 
-        <div class="notifications">
```

Comment:
@AugierLe42e This stuff, a div with `.notifications` class seems to be missing. It it used extensively in the `notifications-badge` file https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/widgets/notifications-badge.js#L94

Also, regarding this file. It contains the "infinite scroll" feature for the notifications. It's not related to the badge at all, it should be moved out of that file. It makes things much more confusing.